paimon-hive/paimon-hive-catalog/src/main/java/org/apache/paimon/hive/HiveCatalog.java

```diff
@@ -471,6 +468,131 @@ public boolean caseSensitive() {
+
+        CompletableFuture<Void> allOf =
+                CompletableFuture.allOf(
+                        tables.stream()
+                                .map(table -> Identifier.create(databaseName, table))
+                                .map(
+                                        identifier ->
+                                                CompletableFuture.runAsync(
+                                                        () -> {
+                                                            try {
+                                                                repairTable(identifier);
+                                                            } catch (TableNotExistException e) {
+                                                                LOG.error(
+                                                                        "Table {} does not exist in the paimon.",
+                                                                        identifier.getFullName());
+                                                                // ignore
+                                                            }
+                                                        },
+                                                        COMMON_IO_FORK_JOIN_POOL))
+                                .toArray(CompletableFuture[]::new));
+        allOf.join();
+    }
+
+    private void repairHmsDatabase(String databaseName) {
+        checkNotSystemDatabase(databaseName);
+        try {
+            Database database = client.getDatabase(databaseName);
+            Path newPath = newDatabasePath(databaseName);
+            if (!locationHelper.getDatabaseLocation(database).equals(newPath.toUri().toString())) {
+                locationHelper.specifyDatabaseLocation(newPath, database);
+                client.alterDatabase(databaseName, database);
+            }
+        } catch (NoSuchObjectException e) {
+            try {
+                createDatabase(databaseName, true);
+            } catch (DatabaseAlreadyExistException e1) {
+                // ignore
+            }
+        } catch (TException e) {
+            throw new RuntimeException(
+                    "Failed to determine if database "
+                            + databaseName
+                            + " exists in hive metastore.",
+                    e);
+        }
+    }
+
+    @Override
+    public void repairTable(Identifier identifier) throws TableNotExistException {
+        checkNotSystemTable(identifier, "repairTable");
+        validateIdentifierNameCaseInsensitive(identifier);
+
+        FileStoreTable paimonTable = getDataTable(identifier, true);
+        validateFieldNameCaseInsensitive(paimonTable.rowType().getFieldNames());
+
+        TableSchema tableSchema = paimonTable.schema();
+        try {
+            try {
+                Table table =
+                        client.getTable(identifier.getDatabaseName(), identifier.getObjectName());
```

Comment:
If the table already exists, we will use the location set in it instead of table in the path.

Please add tests to cover this.

Reply:
done.